### PR TITLE
[WasmFS] Implement __syscall_faccessat

### PR DESCRIPTION
**Pull request by kripken**
Implement `__syscall_faccessat`. It is basically the same as the existing
`__syscall_access` except it has two more flags.

To get the test passing, also:
* Add `FS.chmod`
* Fix `FS.mkdir` which did not receive or sent the mode param to wasm
* Implement `__syscall_fstatat64`

### CODE REVIEW

File: system/lib/wasmfs/syscalls.cpp [hunk 1]
```diff
@@ -1019,4 +1017,8 @@ long __syscall_access(long path, long amode) {
 
   return 0;
 }
+
+long __syscall_access(long path, long amode) {
```
Comment (sbc100) [inline system/lib/wasmfs/syscalls.cpp:1021]: It seems odd that we would want/need to support both of the syscalls...

Comment (sbc100) [reply system/lib/wasmfs/syscalls.cpp:1021]: I looks like we can/should make PR to remove our definition of __syscall_access so musl will just use __syscall_accessat

Comment (kripken) [reply system/lib/wasmfs/syscalls.cpp:1021]: Ok, I'll try that.

**Comment by kripken** [comment]
Ok, after #16296 this is now much simpler. It just:

* Add FS.chmod
* Fix FS.mkdir which did not receive or sent the mode param to wasm
* Test tweaks and enable testing of `test_unistd_access`

**Review by tlively — APPROVED** [review]

**Pull request merged**